source/undostack.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "command.h"
+
+#include <array>
+#include <memory>
+#include <vector>
+
+class UndoStack {
+public:
+    UndoStack() = default;
+    ~UndoStack() = default;
+
+    void push(std::unique_ptr<Command> &cmd);
+
+    void undo();
+    void redo();
+    [[nodiscard]] bool canUndo() const;
+    [[nodiscard]] bool canRedo() const;
+
+    void clear();
+
+private:
+    bool m_canUndo = false;
+    bool m_canRedo = false;
+    int8_t m_undoPos = 0;
+    int8_t m_redoPos = 0;
```

Comment:
Is `m_redoPos` going to be used for anything?

Reply:
Good point, I forgot to delete it, it was here from a previous iteration of my idea. Removed.